tests/iam/permhandler/permhandler_test.cpp

```diff
@@ -0,0 +1,211 @@
+}
+
+TEST_F(PermHandlerTest, GetPermissionsNoMemoryForPerms)
+{
+    const InstanceIdent instanceIdent1 {"serviceID1", "subjectID1", 1};
+
+    FunctionalServicePermissions testServicePermissions;
+    testServicePermissions.mName = "testService";
+    testServicePermissions.mPermissions.PushBack({"key1", "value1"});
+    testServicePermissions.mPermissions.PushBack({"key2", "value2"});
+
+    StaticArray<FunctionalServicePermissions, 2> funcServerPermissions;
+    funcServerPermissions.PushBack(testServicePermissions);
+
+    const auto errRegister = mPermHandler.RegisterInstance(instanceIdent1, funcServerPermissions);
+    ASSERT_TRUE(errRegister.mError.IsNone()) << errRegister.mError.Message();
+
+    const auto secret = errRegister.mValue;
+
+    InstanceIdent                resInstanceIdent;
+    StaticArray<PermKeyValue, 1> resServicePerms;
+
+    const auto errGetPermissions
+        = mPermHandler.GetPermissions(resInstanceIdent, resServicePerms, secret, "testService");
+    ASSERT_TRUE(errGetPermissions.Is(ErrorEnum::eNoMemory)) << errGetPermissions.Message();
+    ASSERT_TRUE(resServicePerms.IsEmpty());
+}
+
+TEST_F(PermHandlerTest, UnregisterInstance)
+{
+    InstanceIdent instanceIdent;
+    instanceIdent.mServiceID = "test-service-id";
+
+    StaticArray<FunctionalServicePermissions, 1> perms;
+
+    auto err = mPermHandler.UnregisterInstance(instanceIdent);
+    EXPECT_FALSE(err.IsNone()) << err.Message();
+
+    const auto errWithValue = mPermHandler.RegisterInstance(instanceIdent, perms);
+    EXPECT_TRUE(errWithValue.mError.IsNone()) << errWithValue.mError.Message();
+
+    err = mPermHandler.UnregisterInstance(instanceIdent);
+    EXPECT_TRUE(err.IsNone()) << err.Message();
+
+    err = mPermHandler.UnregisterInstance(instanceIdent);
+    EXPECT_TRUE(err.Is(ErrorEnum::eNotFound)) << err.Message();
+}
+
+TEST_F(PermHandlerTest, TestInstancePermissions)
+{
+    const InstanceIdent instanceIdent1 {"serviceID1", "subjectID1", 1};
+    const InstanceIdent instanceIdent2 {"serviceID2", "subjectID2", 2};
+    InstanceIdent       instance;
+
+    FunctionalServicePermissions visServicePermissions;
+    visServicePermissions.mName = "vis";
+    visServicePermissions.mPermissions.PushBack({"*", "rw"});
+    visServicePermissions.mPermissions.PushBack({"test", "r"});
+
+    FunctionalServicePermissions systemCoreServicePermissions;
+    systemCoreServicePermissions.mName = "systemCore";
+    systemCoreServicePermissions.mPermissions.PushBack({"test1.*", "rw"});
+    systemCoreServicePermissions.mPermissions.PushBack({"test2", "r"});
+
+    StaticArray<FunctionalServicePermissions, 2> funcServerPermissions;
+    funcServerPermissions.PushBack(visServicePermissions);
+    funcServerPermissions.PushBack(systemCoreServicePermissions);
+
+    auto       errRegister = mPermHandler.RegisterInstance(instanceIdent1, funcServerPermissions);
+    const auto secret1     = errRegister.mValue;
```

Comment:
intermediate var is not required

Reply:
this UT has 2 secret variables (for 2 different instances) that are used in test